Tribler/Test/test_as_server.py

```diff
@@ -130,13 +145,28 @@ def tearDown(self, annotate=True):
         for reader in open_readers:
             self.assertNotIsInstance(reader, BasePort, "The test left a listening port behind: %s" % reader)
 
+    @deferred(timeout=5)
+    def tearDown(self, annotate=True):
+        self.tearDownCleanup()
+        if annotate:
+            self.annotate(self._testMethodName, start=False)
+
         process_unhandled_exceptions()
+        process_unhandled_twisted_exceptions()
+
         self.watchdog.join(2)
         if self.watchdog.is_alive():
             self._logger.critical("The WatchDog didn't stop!")
             self.watchdog.print_all_stacks()
             raise RuntimeError("Couldn't stop the WatchDog")
 
+        if self.file_server:
+            return maybeDeferred(self.file_server.stopListening).addCallback(self.checkReactor)
+        else:
+            deferred = Deferred().addCallback(self.checkReactor)
```

Comment:
Nice trick! (I read in the docs that it requires a `callable` as parameter, so it has to be `succeeded(self.checkReactor)`).